src/natcap/invest/ndr/ndr.py

```diff
@@ -603,61 +604,66 @@ def execute(args):
+            'default_nodata': _TARGET_NODATA,
+        },
+        dependent_task_list=[mask_task, align_raster_task],
         target_path_list=[f_reg['masked_runoff_proxy_path']],
-        task_name='mask runoff proxy raster')
-
-    dem_nodata = pygeoprocessing.get_raster_info(
-        f_reg['aligned_dem_path'])['nodata'][0]
-    dem_target_nodata = float(  # GDAL expects a python float, not numpy.float32
-        numpy.finfo(numpy.float32).min if dem_nodata is None else dem_nodata)
+        task_name='mask runoff proxy raster',
+    )
     mask_dem_task = task_graph.add_task(
-        func=gdal.Warp,
+        func=_mask_raster,
         kwargs={
-            'destNameOrDestDS': f_reg['masked_dem_path'],
-            'srcDSOrSrcDSTab': f_reg['aligned_dem_path'],
-            'outputType': gdal.GDT_Float32,
-            'dstNodata': dem_target_nodata,
-            'cutlineDSName': args['watersheds_path']},
-        dependent_task_list=[align_raster_task],
+            'source_raster_path': f_reg['aligned_dem_path'],
+            'mask_raster_path': f_reg['mask_path'],
+            'target_masked_raster_path': f_reg['masked_dem_path'],
+            'target_dtype': gdal.GDT_Float32,
+            'default_nodata': float(numpy.finfo(numpy.float32).min),
+        },
+        dependent_task_list=[mask_task, align_raster_task],
         target_path_list=[f_reg['masked_dem_path']],
-        task_name='mask dem raster')
-
-    lulc_nodata = pygeoprocessing.get_raster_info(
-        f_reg['aligned_lulc_path'])['nodata'][0]
-    lulc_target_nodata = (
-        numpy.iinfo(numpy.int32).min if lulc_nodata is None else lulc_nodata)
+        task_name='mask dem raster',
+    )
     mask_lulc_task = task_graph.add_task(
-        func=gdal.Warp,
+        func=_mask_raster,
         kwargs={
-            'destNameOrDestDS': f_reg['masked_lulc_path'],
-            'srcDSOrSrcDSTab': f_reg['aligned_lulc_path'],
-            'outputType': gdal.GDT_Int32,
-            'dstNodata': lulc_target_nodata,
-            'cutlineDSName': args['watersheds_path']},
-        dependent_task_list=[align_raster_task],
+            'source_raster_path': f_reg['aligned_lulc_path'],
+            'mask_raster_path': f_reg['mask_path'],
+            'target_masked_raster_path': f_reg['masked_lulc_path'],
+            'target_dtype': gdal.GDT_Int32,
+            'default_nodata': numpy.iinfo(numpy.int32).min,
+        },
+        dependent_task_list=[mask_task, align_raster_task],
         target_path_list=[f_reg['masked_lulc_path']],
-        task_name='mask lulc raster')
+        task_name='mask lulc raster',
+    )
 
     fill_pits_task = task_graph.add_task(
         func=pygeoprocessing.routing.fill_pits,
         args=(
             (f_reg['masked_dem_path'], 1), f_reg['filled_dem_path']),
         kwargs={'working_dir': intermediate_output_dir},
-        dependent_task_list=[align_raster_task],
+        dependent_task_list=[align_raster_task, mask_dem_task],
```

Comment:
I believe that all tasks in `dependent_task_list` must complete before `fill_pits_task` will execute, and so since `mask_dem_task` depends on `align_raster_task`, the order of execution will be `align_raster_task` --> `mask_dem_task` --> `fill_pits_task`.